src/device_manager.rs

```diff
@@ -3,12 +3,116 @@
 
 //! System level device management.
 //!
-//! [IoManager](struct.IoManager.html) is respondsible for managing
+//! [`IoManager`] is respondsible for managing
 //! all devices of virtual machine, registering IO resources callback,
 //! deregistering devices and helping VM IO exit handling.
-//！VMM would be responsible for getting device resource request, ask
-//! vm_allocator to allocate the resources, ask vm_device to register the
-//! devices IO ranges, and finally set resources to virtual device.
+//! It defines two buses, one for PIO and one for MMIO, and provides default
+//! implementations of [`PioManager`] and [`MmioManager`].
+//!
+//! VMM would be responsible for getting device resource request, allocating the
+//! resources, asking `vm_device` to register the devices IO ranges, and finally
+//! setting resources to virtual device.
```

Comment:
I know this phrase wasn't added in this PR, but it sounds a bit strange, e.g. `asking vm_device something`; the `getting device resource request` is also confusing (for me at least). Can we somehow rephrase it please?

Reply:
Please suggest how can we rephrase this.

Reply:
"The VMM must first allocate unique resources (such as bus ranges), and then call into the `vm-device` interface to register the devices with their corresponding resources."
WDYT?

Reply:
Sounds good to me, thanks.